data-access/src/object_store/mod.rs

```diff
@@ -102,3 +132,118 @@ pub trait ObjectStore: Sync + Send + Debug {
     /// Get object reader for one file
     fn file_reader(&self, file: SizedFile) -> Result<Arc<dyn ObjectReader>>;
 }
+
+const GLOB_START_CHARS: [char; 3] = ['?', '*', '['];
+
+/// Determine whether the path contains a globbing character
+fn contains_glob_start_char(path: &str) -> bool {
+    path.chars().any(|c| GLOB_START_CHARS.contains(&c))
+}
+
+/// Filters the file_stream to only contain files that end with suffix
+fn filter_suffix(file_stream: FileMetaStream, suffix: &str) -> Result<FileMetaStream> {
+    let suffix = suffix.to_owned();
+    Ok(Box::pin(file_stream.filter(move |fr| {
```

Comment:
try_filter would perhaps be nicer https://docs.rs/futures/latest/futures/stream/trait.TryStreamExt.html#method.try_filter

Reply:
This is the code that was already in place. 

When I update the code to use try_filter, it will not pass-through the errors anymore (not sure if anyone was actually doing something with them though..)

Reply:
Rebased to handle the merge conflict

Reply:
>  it will not pass-through the errors anymore

It should do? "All errors are passed through without filtering in this combinator."